.gitignore

```diff
@@ -4,6 +4,7 @@
 .config
 .yardoc
 gemfiles/*.lock
+Gemfile.lock
```

Comment:
This PR deleted the Gemfile.lock because right now it's out of date for use by developers and because CI doesn't use it. If you prefer it to remain in the repository then I'll put it back but update it.

Reply:
And by "out of date", I mean that the version of bundler being used is not supported by modern Ruby:

```
/home/flavorjones/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-1.13.7/lib/bundler/shared_helpers.rb:161:in `search_up': undefined method `untaint' for "/home/flavorjones/code/oss/rails-dom-testing":String (NoMethodError)

      current  = File.expand_path(SharedHelpers.pwd).untaint
                                                    ^^^^^^^^
```

Reply:
For a project like this one that aims to support such a wide swatch of Ruby versions, I don't think it's useful to have a lockfile checked in.